source/SharedCrtResourceManager.cpp

```diff
@@ -43,20 +44,46 @@ bool SharedCrtResourceManager::locateCredentials(const PlainConfig &config)
 {
     struct stat fileInfo;
     bool locatedAll = true;
-    if (stat(config.key->c_str(), &fileInfo) != 0)
+    if (config.secureElement.enabled)
     {
-        LOGM_ERROR(TAG, "Failed to find %s, cannot establish MQTT connection", Sanitize(config.key->c_str()).c_str());
-        locatedAll = false;
+        if (stat(config.secureElement.pkcs11Lib->c_str(), &fileInfo) != 0)
+        {
+            LOGM_ERROR(
+                TAG,
+                "Failed to find PKCS#11 library file: %s, cannot establish MQTT connection",
+                Sanitize(config.secureElement.pkcs11Lib->c_str()).c_str());
+            locatedAll = false;
+        }
+        else
+        {
+            string parentDir = FileUtils::ExtractParentDirectory(config.secureElement.pkcs11Lib->c_str());
```

Comment:
Actually only `root-ca` validation is done in Config.cpp file. We should move everything to Config.cpp but maybe in a separate PR since it would require us to update couple of unit tests.

Reply:
Separate PR would be fine but several file permissions are validated in config.cpp: The sensor-publish socket path, parent dir, config file, config file parent dir, root ca, and input/output shadow files.